src/main/java/com/jadventure/game/entities/Entity.java

```diff
@@ -257,6 +239,29 @@ public String getWeapon() {
         return result;
     }
 
+    private EquipmentLocation checkNewEquipmentLocation(EquipmentLocation place, Item item) {
+        if (place == null) {
+            place = item.getPosition();
+        }
```

Comment:
I think it would make sense to move this to the original method. It updates `place` to the given `item`'s equipment slot whereas the rest of the method does the action of unequipping the applicable gear.

This would also remove the need for the return statement.
```suggestion
```